ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -717,10 +763,26 @@ func (r *RayJobReconciler) checkActiveDeadlineAndUpdateStatusIfNeeded(ctx contex
 	if rayJob.Spec.ActiveDeadlineSeconds == nil || time.Now().Before(rayJob.Status.StartTime.Add(time.Duration(*rayJob.Spec.ActiveDeadlineSeconds)*time.Second)) {
 		return false
 	}
-	logger.Info("The RayJob has passed the activeDeadlineSeconds. Transition the status to `Failed`.", "StartTime", rayJob.Status.StartTime, "ActiveDeadlineSeconds", *rayJob.Spec.ActiveDeadlineSeconds)
-	rayJob.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusFailed
+
```

Comment:
The [K8s Job doc](https://kubernetes.io/docs/concepts/workloads/controllers/job/) said that

> Note that a Job's .spec.activeDeadlineSeconds takes precedence over its .spec.backoffLimit. Therefore, a Job that is retrying one or more failed Pods will not deploy additional Pods once it reaches the time limit specified by activeDeadlineSeconds, even if the backoffLimit is not yet reached.

Maybe we should not retry if the RayJob reaches the time limit specified by `activeDeadlineSeconds`.

Reply:
Good catch, I reverted the change for activeDeadlineSeconds, it should now ignore backoffLimit